### PR TITLE
lib: dk_buttons_and_leds: Add interrupt driven button handling

### PR DESCRIPTION
**Pull request by jhn-nordic**
Adds interrupt driven button handling with GPIOTE PORT interrupt to library.

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

**Comment by jhn-nordic** [comment]
Inspired by the button handling logic in nrf_desktop. Use PORT interrupt to save GPIOTE interrupts for the applications.

File: lib/dk_buttons_and_leds/dk_buttons_and_leds.c [hunk 1]
```diff
@@ -145,7 +245,38 @@ int dk_buttons_init(button_handler_t button_handler)
...
+		/* Module starts in scanning mode and will switch to
+		 * callback mode if no button is pressed.
+		 */
+		err = gpio_pin_disable_callback(gpio_dev, button_pins[i]);
+		if (err) {
+			LOG_ERR("Cannot disable callbacks()");
+			return err;
+		}
+
+		pin_mask |= (1 << button_pins[i]);
```
Comment (pdunaj) [inline lib/dk_buttons_and_leds/dk_buttons_and_leds.c]: `BIT(button_pins[i])`

**Comment by jhn-nordic** [comment]
@jtguggedal , @hmhalvorsen Please have a look and we can get it merged If you have no complaints.

**Comment by jtguggedal** [comment]
@jhn-nordic Unfortunately don't have time to test it now, but I've had a look, and to me it looks good to go. I'll just dismiss my review request as I can't review properly but don't have any complaints either.

**Comment by jhn-nordic** [comment]
@pdunaj Feel free to merge it when you have time

**Pull request merged**